Automattic-Tracks-iOS/Crash Logging/SentryEvent+Extensions.swift

```diff
@@ -0,0 +1,33 @@
+import Sentry
+
+extension Event {
+    static func from(
+        error: NSError,
+        level: SentryLevel = .error,
+        user: Sentry.User? = nil,
+        timestamp: Date = Date(),
+        extra: [String: Any] = [:]
+    ) -> Event {
```

Comment:
nit: why not `init(error:level:user:timestamp:extra:)` (or `init(fromError:level:user:timestamp:extra:)`) instead of `static func`? 🤔

Reply:
Just a habit I got into when using Rust – I particularly like it for situations like this where I don't control the `SentryEvent` object – I feel like they own the `init` namespace, so I like static initializers for this case 🤷

Reply:
Right, got it. Kinda makes sense.

It's just that for people used to Swift, I don't think they will think about trying `SentryEvent.from ⇥` to check with autocomplete if there's a way to create a `SentryEvent` from an `NSError`. It's more idiomatic and common to type `SentryEvent  ⇥` or `SentryEvent.init ⇥` to discover all the ways to create a `SentryEvent` by relying on autocomplete, so having it as a `static func` might reduce discoverability here. But 🤷 not that big a deal.